agent/agentcontainers/containers_dockercli.go

```diff
@@ -192,7 +193,10 @@ func devcontainerEnv(ctx context.Context, execer agentexec.Execer, container str
 	env := make([]string, 0)
 	for _, m := range meta {
 		for k, v := range m.RemoteEnv {
-			env = append(env, fmt.Sprintf("%s=%s", k, v))
+			if v == nil { // *string per spec
```

Comment:
I interpret this as env var `k` should be removed. Does the spec specify more concretely what null means in this context (or do we have to try it out)?

Reply:
Unfortunately, no. I think we'll have to either try it out or read the implementation.

Reply:
It seems `@devcontainer/cli` has a poor implementation, from:

```json
	"remoteEnv": {
		"FOO": "bar",
		"NODE_OPTIONS": "",
		"GOPRIVATE": null
	},
```

We get this.

```
NODE_OPTIONS=
GOPRIVATE=null
FOO=bar
```

I'm not a fan of the nut handling here. But if we want to retain a similar behavior, we shouldn't just skip it.

I'd suggest setting it to an empty string when "none", does that sound reasonable? Or do we want to mimic `@devcontainer/cli` exactly here?